src/AzureExtension/Providers/RepositoryProvider.cs

```diff
@@ -152,29 +155,38 @@ private List<IRepository> GetRepos(VssConnection organizationConnection, TeamPro
     /// <returns>A list of repositories the user has access to.</returns>
     public IAsyncOperation<RepositoriesResult> GetRepositoriesAsync(IDeveloperId developerId)
     {
-        if (developerId is not DeveloperId.DeveloperId azureDeveloperId)
+        return Task.Run(() =>
         {
-            return Task.Run(() =>
+            try
             {
-                var exception = new NotSupportedException("Authenticated user is not the an azure developer id.");
-                return new RepositoriesResult(exception, $"{exception.Message} HResult: {exception.HResult}");
-            }).AsAsyncOperation();
-        }
+                if (developerId is not DeveloperId.DeveloperId azureDeveloperId)
+                {
+                    var exception = new NotSupportedException("Authenticated user is not the an azure developer id.");

```

Comment:
>var exception = [](http://example.com/codeflow?start=20&length=15)

isn't it more fun to just throw it and let the catch handler on 185 construct a weird result object?